server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -352,9 +390,13 @@ public void doRun()
           {
             try {
               for (Pair<FireHydrant, Interval> pair : indexesToPersist) {
-                metrics.incrementRowOutputCount(persistHydrant(pair.lhs, schema, pair.rhs));
+                metrics.incrementRowOutputCount(
+                    persistHydrant(
```

Comment:
outstanding review comment --
@gianm -  Will we lose data if one hydrant is persisted with the metadata, then the plumber crashes? If I'm reading the code right, that would cause the next bootstrap to think that all the previously read data was persisted.

@himanshug - hmmm... that sounds correct, still thinking what would be the right thing to do here...
may be create a marker file in the end at /persist_dir/datasource/, and use commit metadata information only if the marker file was present?